src/locking/spinlock.rs

```diff
@@ -9,43 +9,107 @@ use core::fmt::Debug;
 use core::ops::{Deref, DerefMut};
 use core::sync::atomic::{AtomicU64, Ordering};
 
+/// A lock guard used in conjunction with [`SpinLock`]. This lock guard
```

Comment:
I'd make it explicit that one obtains this type from a `SpinLock`:

```suggestion
/// A lock guard obtained from a [`SpinLock`]. This lock guard
```